examples/drop_sub_components.rs

```diff
@@ -0,0 +1,280 @@
+        ComponentParts { model, widgets }
+    }
+
+    fn update_with_view(
+        &mut self,
+        widgets: &mut Self::Widgets,
+        message: Self::Input,
+        sender: ComponentSender<Self>,
+        root: &Self::Root,
+    ) {
+        widgets.container.remove_all();
+        match message {
+            Msg::ShowInitialScreen => {
+                let controller =
+                    InitialScreen::builder()
+                        .launch(())
+                        .forward(sender.input_sender(), |msg| match msg {
+                            InitialScreenOutput::ShowSubScreen1 => Msg::ShowSubScreen1,
+                            InitialScreenOutput::ShowSubScreen2 => Msg::ShowSubScreen2,
+                        });
+                widgets.container.append(controller.widget());
+                self.mode = Some(AppMode::Initial(controller));
+            }
+            Msg::ShowSubScreen1 => {
+                let controller = SubScreen1::builder()
+                    .launch(())
+                    .forward(sender.input_sender(), |_| Msg::ShowInitialScreen);
+                widgets.container.append(controller.widget());
+                self.mode = Some(AppMode::SubScreen1(controller));
+            }
+            Msg::ShowSubScreen2 => {
+                let controller = SubScreen2::builder()
+                    .launch(())
+                    .forward(sender.input_sender(), |_| Msg::ShowInitialScreen);
+                widgets.container.append(controller.widget());
+                self.mode = Some(AppMode::SubScreen2(controller));
+            }
+        }
+        root.set_default_size(400, 300);
+    }
+}
+
+const INITIAL_SCREEN_COMPONENT_NAME: &str = "InitialScreen";
+
+struct InitialScreen {}
+
+#[derive(Debug)]
+enum InitialScreenOutput {
+    ShowSubScreen1,
+    ShowSubScreen2,
+}
+
+#[relm4::component]
+//noinspection RsSortImplTraitMembers
+impl SimpleComponent for InitialScreen {
+    type Input = ();
+    type Output = InitialScreenOutput;
+    type Init = ();
+
+    fn init(
+        _init: Self::Init,
+        root: Self::Root,
+        sender: ComponentSender<Self>,
+    ) -> ComponentParts<Self> {
+        let model = Self {};
+        let widgets = view_output!();
+        ComponentParts { model, widgets }
+    }
+
+    view! {
+        gtk::Box {
+            set_orientation: gtk::Orientation::Vertical,
+            set_hexpand: true,
+            set_vexpand: true,
+            set_halign: gtk::Align::Center,
+            set_valign: gtk::Align::Center,
+            gtk::Box {
+                set_orientation: gtk::Orientation::Vertical,
+                set_spacing: css::COMPONENT_SPACING,
+                gtk::Button {
+                    set_label: "Sub screen 1",
+                    connect_clicked[sender] => move |_| sender.output(InitialScreenOutput::ShowSubScreen1).unwrap_or_else(|e| log_sender_output_error(INITIAL_SCREEN_COMPONENT_NAME, e))
```

Comment:
Using `.unwrap()` or `.expect()` might be ok in this case. If the parent component forgets to forward the output messages, that's a considerable programming error that would justify a panic IMO.